### DISCUSSION TITLE
根据 cmakelists.txt 生成xmake.lua

### DISCUSSION THREAD

**Discussion by kyoybs** [xmake-io/xmake#2850] [Ideas]
### 你在什么场景下需要该功能？

大量类库提供了 cmakelists.txt， 但是没有提供xmake.lua， 这时候想通过cmakelists.txt生成 xmake.lua， 从而转用xmake.

### 描述可能的解决方案

生成 cmakelists.txt的逆向功能.

**Comment by waruqi** [comment]
不支持，你可以直接 add_requires 从 xmake-repo/conan/vcpkg 集成第三方 cmake 库，没有的可以提 pr 到 xmake-repo 收录

或者直接本地源码集成，参考 https://xmake.io/#/zh-cn/package/local_3rd_source_library

原本就是可以直接兼容 cmakelists ，没必要去转

**Comment by waruqi** [comment]
现在可以用 chatgpt 转了。。

<img width="701" alt="image" src="https://user-images.githubusercontent.com/151335/223956454-6ec0061c-bd7f-4694-8124-a5a633f09fae.png">
<img width="836" alt="image" src="https://user-images.githubusercontent.com/151335/223956764-9cf2f3ac-dcd0-474e-85bf-3ef6cd0dff22.png">
